### PR TITLE
Optimize by caching webmention data into memory

### PR DESCRIPTION
**Pull request by ashmaroli**
Results in massive performance gains by bypassing repeated parsing of YAML files :tada:

### CODE REVIEW

**Comment by ashmaroli** [comment]
@nhoizey Will you be able to test your site against this branch to test if there are any performance improvements (without loss of existing renders from this plugin) in real-world sites..?
Thanks :smiley:

**Comment by aarongustafson** [comment]
Wow, that cut things in half with my test site. Will try on a local build of my site momentarily.

**Comment by aarongustafson** [comment]
On my site it cut build time from ~1000s to ~200s. What a huge difference from such a small change!

File: lib/jekyll/webmention_io.rb [hunk 1]
```diff
@@ -300,19 +301,22 @@ def self.log(type, message)
...
 
     # Utility Method
-    # Writes given +data+ as YAML string into +file+ path.
+    # Caches given +data+ to memory and then proceeds to write +data+
+    # as YAML string into +file+ path.
     #
     # Returns nothing.
     def self.dump_yaml(file, data = {})
+      @webmention_data_cache[file] = data
       File.open(file, "wb") { |f| f.puts YAML.dump(data) }
```
Comment (aarongustafson) [inline lib/jekyll/webmention_io.rb]: Is there a way to async this operation so it doesn’t block?

Comment (ashmaroli) [reply lib/jekyll/webmention_io.rb]: Which of the two operations is the blocker..?
Either ways, `:dump_yaml` is called just 3-4 times in a build, so it shouldn't cause any noticeable bottlenecks..

Comment (aarongustafson) [reply lib/jekyll/webmention_io.rb]: I was figuring the disk write might do some blocking. But maybe it doesn’t.

**Pull request merged**